### DISCUSSION TITLE
Compress the same content, but do not get the same MD5 value.

### DISCUSSION THREAD

**Discussion by scc172fly** [icsharpcode/SharpZipLib#796] [Q&A]
Compress the same content, but do not get the same MD5 value. How can I get the same MD5 value?

**Comment by doomsdayer** [comment]
[cfg_activity_gather_task.txt](https://github.com/icsharpcode/SharpZipLib/files/9950324/cfg_activity_gather_task.txt)
[cfg_activity_gather_task.txt](https://github.com/icsharpcode/SharpZipLib/files/9950325/cfg_activity_gather_task.txt)
same issue, these two txts are content equal, but have different md5 after zip
by new FastZip().CreateZip()

**Reply by piksel** [reply]
If you are using ZIP, then there will be additional meta data, like creation time etc. that would make the containing ZIP file have a different file hash. 

If you upload the resulting zip files instead I can point out what the difference between them are.

**Reply by doomsdayer** [reply]
[2.zip](https://github.com/icsharpcode/SharpZipLib/files/9957186/2.zip)
[1.zip](https://github.com/icsharpcode/SharpZipLib/files/9957188/1.zip)
These 2 zips  are created from the 2 txts, by the CreateZip method.
Same bytes count but different md5.
Is there some way to create thiese zips with same md5?

**Reply by piksel** [reply]
Here is the archivediag output for these file:
https://storage.googleapis.com/archivediag.piksel.se/reports/1.zip-1667911006181348.html
https://storage.googleapis.com/archivediag.piksel.se/reports/2.zip-1667911065182501.html

As you can see, the file dates are different, which of course leads to different archive hashes.
When you create the FastZip instance you can pass a static `DateTime` to it to make all entries in newly created archives to use that time as their file times.

**Reply by doomsdayer** [reply]
Thx, this really works~
